EIPS/eip-5568.md

```diff
@@ -0,0 +1,72 @@
+---
+eip: 5568
+title: Required Action Signals Using Revert Reasons
+description: Signal to wallets that an action is needed by returning a custom revert code
+author: Pandapip1 (@Pandapip1)
+discussions-to: https://ethereum-magicians.org/t/eip-5568-revert-signals/10622
+status: Draft
+type: Standards Track
+category: Interface
+created: 2022-08-31
+requires: 140
+---
+
+## Abstract
+
+This EIP introduces a minimalistic machine-readable (binary) format to signal to wallets that an action needs to be taken by the user using a well-known revert reason. This custom revert reason contains just enough data to be extendable by future EIPs and to take in arbitrary parameters (up to 64 kB of data). Example use cases could include approving a token for an exchange, sending an HTTP request, or requesting the user to rotate their keys after a certain period of time to enforce good hygiene.
+
+## Motivation
+
+Oftentimes, a smart contract needs to signal to a wallet that an action needs to be taken, such as to sign a transaction or send an HTTP request to a URL. Traditionally, this has been done by hard-coding the logic into the frontend, but this EIP allows the smart contract itself to request the action.
+
+This means that, for example, an exchange or a market can directly tell the wallet to approve the smart contract to spend the token, vastly simplifying the front-end code.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Custom Revert Reason
+
+To send a signal to a wallet, a smart contract MUST revert with the following [EIP-140](./eip-140.md) revert data:
```

Comment:
I love this proposal. Minor part is not making it MUST but just RECOMMENDED. Including a reason can be expensive for smart contract.